### PR TITLE
Mcorr test

### PR DESCRIPTION
**Pull request by greaa-aws**
### Description
This PR adds tests for metrics correlation, including:
- Testing for invariance of the `mcorr.forward()` output with respect to the level (baseline value) of the input metrics
- Testing that metrics with zero variation are not assigned to events
- Testing for correctness and metrics assignment accuracy against data (`mcorr_data.pt`) with known event structure
- Testing that the `MCorr` module can be compiled via torchScript

Minor changes include:
- Changing event key names to be more informative
- Skip wavelet approximation when the input metrics data are constant
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: tests/ml_models/test_mcorr.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+def test_level_invariance():
+    """
+    Results should not depend on level of metrics.
+    Test by comparing results on randomly shifted data.
+    """
+    test_data = torch.load(TESTDATA_FILENAME)["mcorr_level"]
+    model = mcorr.MCorr()
+
+    M, _ = test_data.shape
+    events_noshift = model.forward(test_data, max_events=5)
```
Comment (dhrubo-os) [inline tests/ml_models/test_mcorr.py:34]: Should we change `max_events: int = 3` in our function definition then? I remember we had a conversation about checking this default setting?

File: tests/ml_models/test_mcorr.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
+# SPDX-License-Identifier: Apache-2.0
```
Comment (dhrubo-os) [inline tests/ml_models/test_mcorr.py:1]: Can we create a folder (`metrics_correlation` may be?) under `tests/ml_models` and then put all the mcorr related test files inside that folder?

File: tests/ml_models/test_mcorr.py (hunk 1 above)
Comment (greaa-aws) [reply tests/ml_models/test_mcorr.py:34]: Yes, I think this is a better choice. I can make that change.

File: opensearch_py_ml/ml_models/metrics_correlation/mcorr.py [hunk 3]
```diff
@@ -40,6 +40,16 @@ def forward(
         :return:
         :rtype: List[int, Dict[str, torch.Tensor]]
         """
+        # Validate inputs
+        M, T = metrics.shape
+        if M >= T:
+            raise ValueError(
+                "The number of metrics to correlate must be smaller than the length of each time series."
+            )
+        if max_events > M:
```
[the remark this answers is not in the document]
Comment (greaa-aws) [reply opensearch_py_ml/ml_models/metrics_correlation/mcorr.py:49]: Good callout - I had missed that change. I'll get rid of this check.

**Review by dhrubo-os — APPROVED** [review]

**Pull request merged**